ntbs-service/Pages/Validation.cshtml.cs

```diff
@@ -0,0 +1,152 @@
+using System;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using ntbs_service.Models;
+using ntbs_service.Models.Validations;
+
+namespace ntbs_service.Pages
+{
+    public class ValidationModel : PageModel
+    {
+        protected ContentResult OnPostValidateProperty(object model, string key, string value)
+        {
+            model.GetType().GetProperty(key).SetValue(model, value);
+            return GetValidationResult(model, key);
+        }
+
+        protected ContentResult OnPostValidateDate(object model, string key, string day, string month, string year)
+        {
+            DateTime? convertedDob;
+            var formattedDate = new FormattedDate() { Day = day, Month = month, Year = year };
+            if (formattedDate.TryConvertToDateTime(out convertedDob)) {
+                model.GetType().GetProperty(key).SetValue(model, convertedDob);
+                return GetValidationResult(model, key);
+            }
+            else
+            {
+                return Content(ValidationMessages.ValidDate);
+            }
+        }
+
+        protected ContentResult GetValidationResult(object model, string key)
+        {
+            if (TryValidateModel(model))
+            {
+                return ValidContent();
+            }
+            else
+            {
+                return Content(ModelState[key].Errors[0].ErrorMessage);
+            }
+        }
+
+        public bool IsValid(string key)
+        {
+            return ModelState[key] == null ? true : ModelState[key].Errors.Count == 0;
+        }
+
+        public ContentResult OnPostValidatePatientProperty(string key, string value)
+        {
+            return OnPostValidateProperty(new PatientDetails(), key, value);
+        }
+
+        public ContentResult OnPostValidatePatientDate(string key, string day, string month, string year)
+        {
+            return OnPostValidateDate(new PatientDetails(), key, day, month, year);
+        }
+
+        public ContentResult OnPostValidateClinicalDetailsProperty(string key, string value)
+        {
+            return OnPostValidateProperty(new ClinicalDetails(), key, value);
+        }
+        public ContentResult OnPostValidateClinicalDetailsDate(string key, string day, string month, string year)
+        {
+            return OnPostValidateDate(new ClinicalDetails(), key, day, month, year);
+        }
+
+        public ContentResult OnPostValidateNotificationSiteProperty(string key, string value)
+        {
+            return OnPostValidateProperty(new NotificationSite(), key, value);
+        }
+
+        public ContentResult OnPostValidateYearComparison(string newYear, int existingYear)
+        {

```

Comment:
new method - the dynamic counterpart (called from vue components) to ValidateYearAgainstOtherYear (used serverside) below. I should probably choose a name that makes it clearer these methods are related...

Similary to when we do the formatted date conversion, we need to have two very similar but separate methods here used for dynamic and serverside validation, but I don't think there is a way around this!

Reply:
I think if we really wanted to, we could have the dynamic method call the server-side one and extract the result from the 
 validation state of the model, like we do above with `GetValidationResult`. Do you think that would be worth it?